gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -580,6 +585,13 @@ public FSDataOutputStream create(
 
     checkOpen();
 
+    customLogger.log(CustomLogger.FSOpType.Create, CustomLogger.FSOpStatus.STARTED, hadoopPath.toString(),
+            Stream.of(new String[][] {
+                    { "permission", permission.toString() },
+                    { "overwrite", overwrite + "" },
+                    { "bufferSize", bufferSize + "" },
+            }).collect(Collectors.toMap(data -> data[0], data -> data[1])));
```

Comment:
This seems like compute/memory-intensive operations that will incur resources cost if audit logging is disable.

Reply:
Good point. We can put this behind a config flag and would only be called once the logging is enabled.